modules/javafx.controls/src/main/java/javafx/scene/control/TreeTableView.java

```diff
@@ -1814,6 +1819,7 @@ public int getVisibleLeafIndex(TreeTableColumn<S,?> column) {
      * something external changes and a sort is required.
      */
     public void sort() {
+        sortingInProgress = true;
```

Comment:
I presume that sorting cannot be called recursively?

Reply:
The `sort()` is called only once for a `TreeTableView`. This method does not perform the actual sorting but it is performed by the sort policy(`DEFAULT_SORT_POLICY`). 
`DEFAULT_SORT_POLICY` initiates the sorting by calling `rootItem.sort()`, which results in call to `TreeItem.doSort()`.
`TreeItem.doSort()` makes the final call to sort the children list, this method gets executed once for the root TreeItem and then for each expanded child TreeItem.